apps/api/src/campaign-application/campaign-application.service.ts

```diff
@@ -1,26 +1,75 @@
 import { UpdateCampaignApplicationDto } from './dto/update-campaign-application.dto'
+import { PrismaService } from '../prisma/prisma.service'
+import { OrganizerService } from '../organizer/organizer.service'
+import { Person } from '@prisma/client'
 
 @Injectable()
 export class CampaignApplicationService {
+  constructor(private prisma: PrismaService, private organizerService: OrganizerService) {}
   async getCampaignByIdWithPersonIds(id: string): Promise<UpdateCampaignApplicationDto> {
     throw new Error('Method not implemented.')
   }
 
-  create(createCampaignApplicationDto: CreateCampaignApplicationDto) {
-    if (
-      !createCampaignApplicationDto.acceptTermsAndConditions ||
-      !createCampaignApplicationDto.transparencyTermsAccepted ||
-      !createCampaignApplicationDto.personalInformationProcessingAccepted
-    ) {
-      throw new BadRequestException('All agreements must be checked')
+  async create(createCampaignApplicationDto: CreateCampaignApplicationDto, person: Person) {
+    try {
+      if (
+        createCampaignApplicationDto.acceptTermsAndConditions === false ||
+        createCampaignApplicationDto.transparencyTermsAccepted === false ||
+        createCampaignApplicationDto.personalInformationProcessingAccepted === false
+      ) {
+        throw new BadRequestException('All agreements must be checked')
+      }
+
+      let organizer = await this.prisma.organizer.findUnique({
+        where: { personId: person.id },
+      })
+
+      if (!organizer) {
+        organizer = await this.organizerService.create({
+          personId: person.id,
+        })
+      }
+
+      const sanitizedData = {
+        campaignName: createCampaignApplicationDto.campaignName.trim(),
+        organizerName: createCampaignApplicationDto.organizerName.trim(),
+        organizerEmail: createCampaignApplicationDto.organizerEmail.trim(),
+        organizerPhone: createCampaignApplicationDto.organizerPhone.trim(),
+        beneficiary: createCampaignApplicationDto.beneficiary.trim(),
+        organizerBeneficiaryRel: createCampaignApplicationDto.organizerBeneficiaryRel.trim(),
+        goal: createCampaignApplicationDto.goal.trim(),
+        history: createCampaignApplicationDto.history?.trim(),
+        amount: createCampaignApplicationDto.amount.trim(),
+        description: createCampaignApplicationDto.description?.trim(),
+        campaignGuarantee: createCampaignApplicationDto.campaignGuarantee?.trim(),
+        otherFinanceSources: createCampaignApplicationDto.otherFinanceSources?.trim(),
+        otherNotes: createCampaignApplicationDto.otherNotes?.trim(),
+        category: createCampaignApplicationDto.category,
+        organizerId: organizer.id,
+      }
+
+      const newCampaignApplication = await this.prisma.campaignApplication.create({
+        data: sanitizedData,
+      })
+
+      return newCampaignApplication
+    } catch (error) {
+      console.error('Error in create():', error)
```

Comment:
It looks like `Logger` (imported from 'nestjs`) is the standard logging used in other services. Let's use that. 

@sashko9807 since I'm kinda new in nestjs and looking at for example CampaignService.createCampaign - there is no explicit logging if create fails. So the question is - do we need to explicitly log or the Logger will log [class][method] Error message, error stack automatically if a prisma error appears?